pkg/ccl/changefeedccl/kvfeed/kv_feed_test.go

```diff
@@ -458,6 +465,66 @@ func (f rawEventFeed) run(
 	return nil
 }
 
+// TestDistSenderAllRangeSpans tests (*distserver).AllRangeSpans.
+func TestDistSenderAllRangeSpans(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	const nodes = 1
+	ctx := context.Background()
+	tc := testcluster.StartTestCluster(t, nodes, base.TestClusterArgs{})
+	defer tc.Stopper().Stop(ctx)
+
+	node := tc.Server(0)
+	sqlDB := sqlutils.MakeSQLRunner(tc.ServerConn(0))
+	distSender := node.DistSenderI().(*kvcoord.DistSender)
+
+	tenantPrefix := ""
+	if tc.StartedDefaultTestTenant() {
+		tenantPrefix = "/Tenant/10"
+	}
+
+	// Use manual merging/splitting only.
+	tc.ToggleReplicateQueues(false)
+
+	mergeRange := func(key interface{}) {
+		err := node.DB().AdminMerge(ctx, key)
+		if err != nil {
+			if !strings.Contains(err.Error(), "cannot merge final range") {
+				t.Fatal(err)
+			}
+		}
+	}
+	getTableSpan := func(tableName string) roachpb.Span {
+		desc := desctestutils.TestingGetPublicTableDescriptor(
+			node.DB(), node.Codec(), "defaultdb", "a")
+		return desc.PrimaryIndexSpan(node.Codec())
+	}
+	getTableDesc := func(tableName string) catalog.TableDescriptor {
+		return desctestutils.TestingGetPublicTableDescriptor(
+			node.DB(), node.Codec(), "defaultdb", "a")
+	}
+
+	// Regression test for the issue in #117286.
+	t.Run("returned range does not exceed input span boundaries", func(t *testing.T) {
+		// Merge 3 tables into one range.
+		sqlDB.Exec(t, "create table a (i int primary key)")
+		sqlDB.Exec(t, "create table b (j int primary key)")
+		sqlDB.Exec(t, "create table c (k int primary key)")
+		mergeRange(getTableSpan("a").Key)
+		mergeRange(getTableSpan("b").Key)
+
+		bTableID := getTableDesc("b").GetID()
+		rangeSpans, err := AllRangeSpans(ctx, distSender, []roachpb.Span{getTableSpan("b")})
+		require.NoError(t, err)
+
+		// Assert that the returned range is trimmed, so it only contains spans from "b" and not the entire
+		// range containing "a" and "c".
+		require.Equal(t, 1, len(rangeSpans), fmt.Sprintf("%v", rangeSpans))
+		require.Equal(t, fmt.Sprintf("%s/Table/%d/{1-2}", tenantPrefix, bTableID), rangeSpans[0].String())
```

Comment:
I confirmed that the test fails with the wrong span without the production changes. The output is  `/{Table/62-Max}`